### ISSUE TITLE
training visualization with TensorBoard

### ISSUE THREAD

**Issue by sammlapp** [kitzeslab/opensoundscape#486]
does this require any change inside opso?

**Comment by sammlapp** [comment]
or WandB

**Comment by sammlapp** [comment]
WandB works nicely. For the most part, it can be used in a training or prediction script. However, we probably want to enable some internal support in opso, or refactor training and prediction in a way that makes it possible to log things to wandb without internally supporting it.

Training
- log examples of tensor samples: this can be in user's training script
- log metrics on training and validation
- score histograms would be nice
- log loss

Prediction
- log progress on total files or batches (tqdm?)
- optionally export top-scoring clips! (You could review top-scoring clips in real-time, and could even do simple top-down listening right from wandb interface! however we only have 100 gigs of storage)


Pytorch Lightening provides support for integration with WandB, which we could consider using. I'm skeptical because Pytorch Lightening seems to still be in development and is not very well documented or stable so far.

**Comment by carlybatist** [comment]
Is there a way to use tqdm with model.train or model.predict? I'm not sure where to put in the command since tqdm runs with for loops but the loop is 'behind the scenes' for model.predict.

**Comment by sammlapp** [comment]
@carlybatist good question, sorry I didn't see this question earlier. Since the training loop is internal, the simplest way to use tqdm would be to only train one epoch at a time and use tqdm on a for loop which loops the 1-epoch .train() command. You could do something similar by predicting on one small set of recordings (eg, recordings from one device) and using tqdm on a for loop that loops through each subset of recordings. For instance
```
for epoch in tqdm(num_epochs):
    model.train(.... , epochs=1)
```
or
```
for i,set_of_files in tqdm(enumerate(all_sets_of_files)):
    scores, _, _ = model.predict(set_of_files,...)
    scores.to_csv(f"preds_{i}.csv")
```
